libraries/AP_Motors/AP_MotorsHeli_Dual.h

```diff
@@ -105,34 +99,35 @@ class AP_MotorsHeli_Dual : public AP_MotorsHeli {
     // update_motor_controls - sends commands to motor controllers
     void update_motor_control(RotorControlState state) override;
 
-    // calculate_roll_pitch_collective_factors - calculate factors based on swash type and servo position
-    void calculate_roll_pitch_collective_factors () override;
+    // calculate_swashplate_tilt - calculate tilt of each swashplate based on configuration
+    float get_swashplate (int8_t swash_num, int8_t swash_axis, float pitch_input, float roll_input, float yaw_input, float coll_input);
 
     // move_actuators - moves swash plate to attitude of parameters passed in
     void move_actuators(float roll_out, float pitch_out, float coll_in, float yaw_out)  override;
 
     //  objects we depend upon
     AP_MotorsHeli_RSC           _rotor;             // main rotor controller
+    AP_MotorsHeli_Swash        _swashplate1;        // swashplate1
+    AP_MotorsHeli_Swash        _swashplate2;        // swashplate2
+    SwashInt16Param            _swash1_H3;          // H3 servo positions for swash 1
```

Comment:
related to my other comments perhaps but I guess there's a technical reason why swashplate1 and swashplate2 can't hold their own parameters?  At least if I understand this correctly, it seems like _swash1_H3 really belongs within swashplate1.